spark/src/main/scala/io/delta/tables/DeltaMergeBuilder.scala

```diff
@@ -487,6 +487,7 @@ object DeltaMergeMatchedActionBuilder {
  *
  * @since 0.3.0
  */
+@DeveloperApi
```

Comment:
Since it's for developers and is part of the public API, I thought it should be marked so. You're right it's not used and I was mistaken it was (as I was switching between Spark's and DL's sources).

I can take it out but think we should follow Spark's footsteps and use `@DeveloperApi` to mark classes as the developer API. WDYT?

Reply:
This is a user-facing API that is guaranteed to remain stable in future versions, so not fitting for `@DeveloperApi`. Users are expected to call this API for all insert actions in MERGE via `whenNotMatched()`:
```
deltaTable
  .merge(source, condition = "t.id = s.id")
  .whenNotMatched().insertAll()
  .execute()
```

Reply:
`@DeveloperApi1` is used only for public apis that are not so stable and used only by low level developers who are willing to take the risk. This entire class is super stable API, so the annotation is not needed.

Reply:
Fixed